File: solutions/c/secret-handshake/2/secret_handshake.c
#include "secret_handshake.h"

const char *const actions[] = {"wink", "double blink", "close your eyes", "jump"};

const char **commands(size_t number){
    bool reverse = (number >> 4) & 1;
    int action = 0;
    const char **handshake = calloc(4, sizeof(char *));
    for (int i = 0; i < 4; i++)
        if (number >> ((reverse) ? 3 - i : i) & 1)
            handshake[action++] = actions[(reverse) ? 3 - i : i];
    return handshake;
}

File: solutions/julia/secret-handshake/2/secret-handshake.jl
const actions = ["wink", "double blink", "close your eyes", "jump"]

function secret_handshake(code)
    reversed = Bool((code >> 4) & 1)
    [actions[(reversed) ? 4 - i : i + 1] for i in 0:3 if code >> ((reversed) ? 3 - i : i) & 1 == 1]
end